### PR TITLE
Introduced bulk edition on combinations

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | This contribution add a way to manage update on multiples combinations linked to a product |
| Type? | new feature |
| Category? | Back Office |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-469 |
| How to test? | Select some combinations using checkboxes, use the _bulk actions_ form to apply some new data. Also, you can delete combinations using "delete combinations" buttons. |

bulk_combinations
- [x] Edition of combinations OK
- [x] Deletion of combinations OK
- [x] bulk checkboxes toggle OK
- [ ] Image input (behavior need to be defined)

### CODE REVIEW

File: admin-dev/themes/default/js/bundle/product/product-bulk-combinations.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          values.push({'id' : input.id, 'value': input.value});
+        }
+      });
+
+      return values;
+    }
+  };
+})();
+
+class Combination {
```
Comment (Shudrum) [inline admin-dev/themes/default/js/bundle/product/product-bulk-combinations.js]: I don't think this file is bundle into WebPack with the Babel plugin…

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -385,6 +385,11 @@ public function formAction($id, Request $request)
             ->add('step6', 'PrestaShopBundle\Form\Admin\Product\ProductOptions')
             ->getForm();
 
+        $formBulkCombinations = $this->createForm('PrestaShopBundle\Form\Admin\Product\ProductCombinationBulk', null, [
```
Comment (Shudrum) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: Now we use `array(`only.

File: src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            'label' => $this->translator->trans('Minimum quantity', [], 'AdminProducts'),
+        ));
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(array(
+            'validation_groups' => false,
+            'iso_code' => '',
+            'price_display_precision' => ''
```
Comment (Shudrum) [inline src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php]: Missed that one coma ^^

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combinations.html.twig [hunk 4]
```diff
@@ -27,11 +28,28 @@
     </div>
   </div>
 
+  <div id="combinations-bulk-form">
+    <div class="row">
+      <p data-toggle="collapse" href="#bulk-combinations-container" aria-expanded="false" aria-controls="bulk-combinations-container">
+        <b>{{ trans('Bulk actions', {}, 'AdminProducts') }}</b>
+        <i class="material-icons pull-right">keyboard_arrow_down</i>
+      </p>
+      <div class="col-md-12 collapse" style="border:2px solid #bbcdd2;" id="bulk-combinations-container">
```
Comment (Shudrum) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combinations.html.twig]: The style should not be here.

Maybe a PrestaShopUI Kit should be added, or a specific component on the SCSS.

File: src/PrestaShopBundle/Resources/views/Admin/Product/form.html.twig [hunk 5]
```diff
@@ -1010,6 +1010,7 @@
   <script src="{{ asset('themes/default/js/bundle/product/product-category-tags.js') }}"></script>
   <script src="{{ asset('themes/default/js/bundle/product/default-category.js') }}"></script>
   <script src="{{ asset('themes/default/js/bundle/product/product-combinations.js') }}"></script>
+  <script src="{{ asset('themes/default/js/bundle/product/product-bulk-combinations.js') }}"></script>
```
Comment (Shudrum) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form.html.twig]: Yep, the `class` will not work.

File: src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig [hunk 6]
```diff
@@ -0,0 +1,70 @@
+{{ form_start(form, {'action': '#', 'method': 'GET'}) }}
+<div class="col-md-12">
+  <div class="col-md-2">
```
Comment (Shudrum) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig]: To use  a col-md-x on another col-md-x, you need to re-set a row:

```html
<div class="col-md-12">
  <div class="row">
    <div class="col-md-2">
    </div>
  </div>
</div>
```

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combinations.html.twig (hunk 4 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combinations.html.twig]: note the design is not part of this pull request, I prefer the inline css here because it will be more easy for @nihco2 to remove it. Do you want me to remove it ?

File: src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig (hunk 6 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig]: wow, I don't want to create a complete tree of div only to re order few blocks !
Do we have alternative "clean" solution ?

Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig]: I've changed, replacing the first "col-md-12" by "row". Thank you for noticing me :)

File: src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php (hunk 3 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php]: done

File: src/PrestaShopBundle/Controller/Admin/ProductController.php (hunk 2 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: done

File: src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig [hunk 7]
```diff
@@ -1,4 +1,4 @@
-{{ form_start(form, {'action': '#', 'method': 'GET'}) }}
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form_combinations_bulk.html.twig]: @Shudrum couldnt keep a real form to preserve footer :/

**Comment by vTerenti** [comment]
QA OK

**Comment by Shudrum** [comment]
👍  We just need to add the script into WebPack before merging this PR. I'm on it.
